test/jdk/java/nio/jni/NewDirectByteBuffer.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+import java.nio.ByteBuffer;
+
+/*
+ * @test
+ * @bug 8299684
+ * @summary Verify that JNI NewDirectByteBuffer throws IllegalArgumentException
+ * if the capacity is negative or greater than Integer::MAX_VALUE
+ * @run main/native NewDirectByteBuffer
+ */
+public class NewDirectByteBuffer {
+    static {
+        System.loadLibrary("NewDirectByteBuffer");
+    }
+
+    private static final long[] ILLEGAL_CAPACITIES = {
+        (long)Integer.MIN_VALUE - 1L,
+        -1L,
+        (long)Integer.MAX_VALUE + 1L,
+        3_000_000_000L,
+        5_000_000_000L
+    };
+
+    private static final long[] LEGAL_CAPACITIES = {
+        0L,
+        1L,
+        (long)Integer.MAX_VALUE - 1,
+        (long)Integer.MAX_VALUE
```

Comment:
Won't such large capacities trigger OOME?

Reply:
> Won't such large capacities trigger OOME?

The malloc might fail, in which case the value of NULL will be passed to NewDirectByteBuffer. It needs to check the return from malloc.

Reply:
Fixed in d2386a4b73f0fa4eb9cc278b3cbb1578e6cb4b55.